src/main/java/com/lina/airline/filter/JwtFilter.java

```diff
@@ -0,0 +1,58 @@
+package com.lina.airline.filter;
+
+import com.lina.airline.service.impl.JWTServiceImp;
+import com.lina.airline.service.impl.CustomUserDetailsServiceImp;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.AllArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.ApplicationContext;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+@RequiredArgsConstructor
+@Component
+@AllArgsConstructor
+public class JwtFilter extends OncePerRequestFilter {
+    private CustomUserDetailsServiceImp userDetailsService;
+    private JWTServiceImp jwtService;
+    ApplicationContext context;
+
+    public JwtFilter(JWTServiceImp jwtService) {
```

Comment:
Already annotated with RequiredArgsConstructor and AllArgsContructor. This constructor should not declare again. But You may get a issue at there. B'coz you have to declare the jwtTokenService as final

Reply:
Fixed this issue.